### ISSUE TITLE
Search inconsistency

### ISSUE THREAD

**Issue by IevgenSobko** [WDaan/VueTorrent#250]
The search field doesn't work as I expect.
For instance, I have the torrent that starts with `True Gift`.

When I type `True` in the filter field I am getting the following
![2021-05-01__11-19-58](https://user-images.githubusercontent.com/14232100/116776184-c64da080-aa6f-11eb-9cff-0dcf632e85b5.png)
From what I see only last item contains the word `True` and I would expect only it to be displayed.
Or maybe it uses some fuzzy search with some mistyping allowance so it finds additional items?

**Comment by WDaan** [comment]
<img width="1012" alt="Screenshot 2021-05-01 at 11 11 37" src="https://user-images.githubusercontent.com/25707052/116777269-00b63e00-aa6e-11eb-924e-9565d31ffd32.png">

Yes, the search uses Fuse.js with a fuzzy matching [threshold](https://fusejs.io/api/options.html#threshold) of 0.3 on the properties listed above. I have little control over what gets matched an what doesn't. I can tweak the threshold a little bit.

**Comment by IevgenSobko** [comment]
Ooops, and my second comment not relevant cause I didn't manage to see the correct torrent name (especially second word).

Lets try to lower `threshold` to get better matching.
Plus there is a `distance` option that we can use with lower values like 2-5 to allow matching no to search too far away in a string.

**Comment by IevgenSobko** [comment]
The current search is a little bit too wide.
If I search `Road` and have category named Downloads it is matching every torrent from this category(about  74).
Because `Downloads` contains the last 3 letters of `Road` and almost all torrents somewhere in mentioned searched fields have the letter `R`.

**Comment by IevgenSobko** [comment]
@WDaan I have played with different options to better understand algorithm and found out that 0.25 threshold works for my needs and I think for most users needs as well 
```js
      const options = {
        threshold: 0.25,
        shouldSort: false,
        keys: ['name', 'size', 'state', 'hash', 'savePath', 'tags', 'category']
      } 
```
Learned a lot about the algorithm and found out that all my above assumptions were wrong and `distance` actually represents how far from the `location` parameter it will match. But it also interconnected with the `threshold` parameter that is multiplied by `distance` to calculate the real limit.
So in this case we have 0,25 * 100*(default distance) = 25 characters from 0(default location) position in a string.
I think it is ok for most torrent names.
Also double-checked hash to make sure its 40 character length won't be ignored by this 25 characters from 0 position. And it seems to work ok for me. So these 25 characters represent the initial position of the match and if a matched part in a string exceeds 25 but matched starting position is within [0, 25] range it will match the whole searched word.

As a result of all stated above could you pls apply a new threshold in the next release?

**Comment by WDaan** [comment]
Thanks for the detailed analysis, I'll make a release with the tweaked value 🙂

**Comment by IevgenSobko** [comment]
Wait a sec I have noticed that double click doesn't wok in master.
Do u see the same behavior?

**Comment by m4ximuel** [comment]
@IevgenSobko It's my mistake, I fixed it on #249

**Comment by WDaan** [comment]
There are also some console.log statements, my bad for not checking thoroughly enough.
But those doesn't matter that much though.
